starter-code/InvestingPortfolio/app/src/main/java/drewmahrt/generalassemb/ly/investingportfolio/MainActivity.java

```diff
@@ -44,9 +53,81 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportLoaderManager().initLoader(LOADER_STOCK,null,this);
 
+        mFAB = (FloatingActionButton) findViewById(R.id.fab);
+        mFAB.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+                LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
+                View dialogPrompt = inflater.inflate(R.layout.dialog_prompt, null);
+                builder.setView(dialogPrompt);
+
+                mNameEditText = (EditText) findViewById(R.id.stock_symbol);
+                mQuantityEditText = (EditText) findViewById(R.id.quantity);
+
+                builder.setPositiveButton("ADD", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int i) {
+                        String name = mNameEditText.getText().toString();
+                        mQuantity = mQuantityEditText.getText().toString();
+                        addStock(name, mQuantity);
+                    }
+                })
+                        .setNegativeButton("CANCEL", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialogInterface, int i) {
+                                Toast.makeText(MainActivity.this, "You Clicked Cancel!",
+                                        Toast.LENGTH_SHORT).show();
+                            }
+                        });
+                AlertDialog dialog = builder.create();
+                dialog.show();
+            }
+        });
+    }
+
+    public void addStock(String stock, String quantity){
+        RequestQueue queue = Volley.newRequestQueue(this);
+
+        JsonArrayRequest jsonArrayRequest = new JsonArrayRequest(
+                Request.Method.GET, BASE_URL + stock, null,
+                new Response.Listener<JSONArray>() {
+                    @Override
+                    public void onResponse(JSONArray response) {
+                        try {
+                            JSONObject object = response.getJSONObject(0);
+                            mSymbol = object.getString("Symbol");
+                            mCompanyName = object.getString("Name");
+                            mExchange = object.getString("Exchange");
+
+                            Log.d(TAG, "onResponse: " + mSymbol);
+                            Log.d(TAG, "onResponse: " + mCompanyName);
+                            Log.d(TAG, "onResponse: " + mExchange);
+
+                            ContentValues values = new ContentValues();
+                            values.put(StockPortfolioContract.Stocks.COLUMN_STOCKNAME, mCompanyName);
+                            values.put(StockPortfolioContract.Stocks.COLUMN_QUANTITY, mQuantity);
```

Comment:
Once you have the `values` ready, you need to use a content resolve to send the `values` to the URI for your ContentProvider